diff-containers/README.md

```diff
@@ -1 +1,238 @@
-# diff-containers
\ No newline at end of file
+# diff-containers
+
+This package implements difference datatypes for container types. Currently,
+there is only one such implemented difference datatype for `Data.Map.Strict`.
+
+## Differences on `Data.Map`s
+
+A difference datatype for maps is defined and implemented in the
+`Data.Map.Diff.Strict` module. A key property of our definition of diffs is that
+diffs can not only be combined/summed, they can also be inverted and summed
+together, which facilitates *subtraction* of diffs. Our definition of diffs is
+not only a `Semigroup` and a `Monoid`, it is also a `Group`.
+
+The goals for our definition of diffs are twofold:
+1. Provide definitions that adhere to the type class laws for `Semigroup`,
+   `Monoid` and `Group`.
+2. Applying diffs to maps should have a nice semantics.
+
+An overview of the sections in this document:
+
+* Section [1) High-level overview](#high-level-overview) provides a high-level
+  overview of diffs, how to create them, and how to apply them.
+* Section [2) A first attempt at implementing
+  diffs](#2-a-first-attempt-at-implementing-diffs) describes a straightforward
+  way to implement diffs, and where this goes wrong if we try to define a
+  `Group` instance.
+* Section [3) Implementing diffs with
+  inverses](#3-implementing-diffs-with-inverses) describes how we implement
+  diffs to facilitate a `Group` instance.
+* Section [4) Applying diffs with inverses: positivity and
+  normality](#4-applying-diffs-with-inverses-positivity-and-normality) describes
+  how applying diffs can fail, but which can be solved by guaranteeing the
+  positivity and normality properties.
+
+### 1) High-level overview
+
+For each of the key-value pairs in a map, the difference type will keep track of
+whether a value is to be inserted or deleted:
```

Comment:
We should think how to prevent the code snippets presented here from bitrotting. In general we should try to avoid writing any code, but that's not always possible :) Maybe we could add a checklist to the PR template that urges the author to check the docs? Or we could move some of these explanations into a literate haskell file, or something like `*.Tutorial.hs`. All in all, these activities are outside the scope of this PR, so please dismiss this comment after reading :)

Reply:
See https://github.com/input-output-hk/anti-diffs/issues/25